text/0101-revamping-the-sdk-performance-api.md

```diff
@@ -0,0 +1,268 @@
+| `op`              | Operation type of the span                                   | String          | Use is discouraged but kept for backwards compatibility for product features                                                                                                                                                                                       |
+| `status`          | Status of the span                                           | String          | An optional final status for this span. Can have three possible values: 'ok', 'error', 'unset'. Same as OpenTelemetry's Span Status                                                                                                                                |
+| `attributes`      | A set of attributes on the span.                             | Object          | This maps to `span.data` in the current schema for spans. There is no existing mapping for this in the current transaction schema. The keys of attributes are well known values, and defined by a combination of OpenTelemtry's and Sentry's semantic conventions. |
+| `measurements`    | Measurements which holds observed values such as web vitals. | Object          |                                                                                                                                                                                                                                                                    |
+
+For the purposes of this RFC, the version on the span schema will be set to 2. This will indicate to all consumers that the new span schema is being used.
+
+Just like both the old Sentry schema and the OpenTelemetry schema, we keep the same fields for `span_id`, `trace_id`, `parent_span_id`, `start_timestamp`, and `end_timestamp`. We also choose to rename `description` to `name` to match the OpenTelemetry schema.
+
+Having both the `name` and `op` fields is redundant, but we choose to keep both for backwards compatibility. There are many features in the product that are powered by having a meaningful operation, more details about this can be found in the documentation around [Span operations](https://develop.sentry.dev/sdk/performance/span-operations/). In the future, we can choose to deprecate `op` and remove it from the schema.
+
+The most notable change here is to formally introduce the `attributes` field, and remove the `span.data` field. This is a breaking change, but worth it in the long-term. If we start accepting `attributes` on transactions as well, we more closely align with the OpenTelemetry schema, and can use the same conventions for both spans and transactions.
```

Comment:
Can we keep the `data` that is gonna be deprecated and forwarded to `attributes` for compatibility?
eg
`span.setData(x, y)` just calls `span.setAttribute(x, y)` internally.

Reply:
Yup this is a good idea! I will also put more details into migrating between the different span schemas.